queryparams/params.go

```diff
@@ -249,38 +247,118 @@ func CalculatePageNumber(offset, pageSize int) int {
 	return ((offset + 1) / pageSize) + 1
 }
 
-type Sort int
+type SortKind int
 
 const (
-	Invalid Sort = iota
-	RelDateAsc
-	RelDateDesc
-	TitleAZ
-	TitleZA
+	InvalidSort SortKind = iota
+	PublishedNewest
+	PublishedOldest
+	UpcomingNewest
+	UpcomingOldest
+	CancelledNewest
+	CancelledOldest
```

Comment:
Do we need the cancelled ones? Aren't we only checking upcoming and non-upcoming (published)?